index.js

```diff
@@ -0,0 +1,529 @@
+
+		return idx;
+	};
+
+	let index = elasticlunrNoStem(function () {
+		this.addField('title');
+		this.addField('description');
+		this.addField('members');
+		this.addField('memberDescriptions');
+		this.setRef('id');
+		this.saveDocument(false);
+	});
+
+	console.log('Generating search index...');	// eslint-disable-line no-console
+
+	readdirp({root: 'src/pages/docs/modules', fileFilter: '*.json'}, (err, res) => {
+		if (!err) {
+			res.files.forEach(result => {
+				const filename = result.fullPath;
+				const json = jsonfile.readFileSync(filename);
+				try {
+					const doc = jsonata(expression).evaluate(json);
+					// Because we don't save the source data with the index, we only have access to
+					// the ref (id). Include both the human-readable title and the path to the doc
+					// in the ref so we can parse it later for display.
+					doc.id = `${doc.title}|docs/modules/${doc.title}`;
+					index.addDoc(doc);
+				} catch (ex) {
+					console.log(chalk.red(`Error parsing ${result.path}`));	// eslint-disable-line no-console
+					console.log(chalk.red(ex));	// eslint-disable-line no-console
+				}
+			});
+		} else {
+			console.error(chalk.red('Unable to find parsed documentation!'));	// eslint-disable-line no-console
+			process.exit(2);
+		}
+
+		readdirp({root: 'src/pages/', fileFilter: '*.md'}, (_err, _res) => {
+			if (!_err) {
+				_res.files.forEach(result => {
+					const filename = result.fullPath;
+					const data = matter.read(filename);
+					const title = data.data.title || pathModule.parse(filename).name;
+					const id = `${title}|${pathModule.relative('src/pages/', pathModule.dirname(filename))}`;
+
+					try {
+						index.addDoc({id, title, description: data.content});
+					} catch (ex) {
+						console.log(chalk.red(`Error parsing ${result.path}`));	// eslint-disable-line no-console
+						console.log(chalk.red(ex));	// eslint-disable-line no-console
+					}
+				});
+				makeDataDir();
+				jsonfile.writeFileSync(docIndexFile, index.toJSON());
+			} else {
+				console.error(chalk.red('Unable to find parsed documentation!'));	// eslint-disable-line no-console
+				process.exit(2);
+			}
+		});
+	});
+	generateLibraryDescription();
+}
+
+function makeDataDir () {
+	mkdirp.sync(dataDir);
+}
+
+function generateLibraryDescription () {
+	const exportContent = JSON.stringify(libraryDescription);
+	makeDataDir();
+	// generate a json file that contains the description to the corresponding libraries
+	fs.writeFileSync(libraryDescriptionFile, exportContent, {encoding: 'utf8'});
+}
+
+/**
+ * Generates a json file with the current Enact version number as read from raw/enact/package.json
+ * TODO: Generate this for all scanned packages
+ */
+function generateDocVersion () {
+	const packageInfo = jsonfile.readFileSync('raw/enact/package.json');
+	const version = JSON.stringify({docVersion: packageInfo.version});
+	makeDataDir(); // just in case
+	fs.writeFileSync(docVersionFile, version, {encoding: 'utf8'});
+}
+
+/**
```

Comment:
It might make sense to move this into a separate bin file referenced from `package.json` so it's not executed when requiring.  Dunno!